src/os/bluestore/BitAllocator.cc

```diff
@@ -1669,3 +1707,11 @@ void BitAllocator::free_blocks_dis(int64_t num_blocks, ExtentList *block_list)
   alloc_assert(get_used_blocks() >= 0);
   unlock();
 }
+
+void BitAllocator::dump()
+{
+  int count = 0;
+  serial_lock(); 
+  dump_int(count);
```

Comment:
humm, it will be quite confusing to implement no nested traversal function. dump_int is to ensure all internal structures shares same code. The count is required because zones do not keep their relative position or their number.